src/components/MonthCalendar.spec.tsx

```diff
@@ -0,0 +1,133 @@
+            <MonthCalendar events={ dummyEvents } month={ dummyMonth } year={ dummyYear } />,
+        );
+        cy.get('[data-testid="calendar-wrapper"]').should('have.css', 'display', 'grid');
+        cy.get('[data-testid="calendar-wrapper"]').children().should('have.length', 42);
+        cy.get('[data-testid="griditem-0"]').then(el => {
+            const topLeft = el[0].getBoundingClientRect().left;
+            cy.get('[data-testid="griditem-6"]').then(el => {
+                const topRight = el[0].getBoundingClientRect().left;
+                cy.get('[data-testid="griditem-35"]').then(el => {
+                    const bottomLeft = el[0].getBoundingClientRect().left;
+                    cy.get('[data-testid="griditem-41"]').then(el => {
+                        const bottomRight = el[0].getBoundingClientRect().left;
+                        expect(topLeft).to.eq(bottomLeft);
+                        expect(topRight).to.eq(bottomRight);
+                    });
+                });
+            });
+        });
+    });
+
+    it('displays the correct date on each grid square', () => {
+        mountWithProviders(
+            <MonthCalendar events={ dummyEvents } month={ dummyMonth } year={ dummyYear } />,
+        );
+        cy.get('[data-testid="griditem-5"]').contains('01');
+        cy.get('[data-testid="griditem-35"]').contains('31');
+    });
+
+    it('displays the correct events on the corresponding date', () => {
+        mountWithProviders(
+            <MonthCalendar events={ dummyEvents } month={ dummyMonth } year={ dummyYear } />,
+        );
+        cy.get('[data-testid="griditem-5"]').contains('01');
+        cy.get('[data-testid="griditem-35"]').contains('31');
+    });
+
+    it('shows out of range dates with another colour', () => {
+        mountWithProviders(
+            <MonthCalendar events={ dummyEvents } month={ dummyMonth } year={ dummyYear } />,
+        );
+
+        cy.get('[data-testid="griditem-4"]').then(el => {
+            const beforeRangeBackgroundColor = el[0].style.backgroundColor;
+            cy.get('[data-testid="griditem-36"]').then(el => {
+                const afterRangeBackgroundColor = el[0].style.backgroundColor;
+                cy.get('[data-testid="griditem-20"]').then(el => {
+                    const inRangeBackgroundColor = el[0].style.backgroundColor;
+                    expect(inRangeBackgroundColor).to.not.eq(beforeRangeBackgroundColor);
+                    expect(afterRangeBackgroundColor).to.eq(beforeRangeBackgroundColor);
+                });
+            });
+        });
+    });
+
+    it('works with a month exactly 4 weeks long starting on Monday', () => {
+        dummyMonth = 1;
+        dummyYear = 2021;
+        mountWithProviders(
+            <MonthCalendar events={ dummyEvents } month={ dummyMonth } year={ dummyYear } />,
+        );
+        cy.get('[data-testid="griditem-0"]').contains('01');
+        cy.get('[data-testid="griditem-27"]').contains('28');
```

Comment:
Oh I see why you would want these test IDs now. Maybe it makes sense to leave them as is (ignore my previous comment)?